pyccel/ast/internals.py

```diff
@@ -163,3 +162,46 @@ def __str__(self):
             stop = str(self.stop)
         return '{0} : {1}'.format(start, stop)
 
+class Symbol(Basic):
+    """
+        #TODO
+    """
+    is_Symbol = True
```

Comment:
What is this for?

Reply:
sometime we is   X._is_Symbol in the code

Reply:
I could only find `is_Symbol` [here](https://github.com/pyccel/pyccel/blob/master/pyccel/ast/core.py#L263) and [here](https://github.com/pyccel/pyccel/blob/master/pyccel/ast/core.py#L269) in the function `allocatable_like`, but I couldn't find any uses of that function. Did you find it anywhere else?
If not then it looks like most of that function is broken, and the docstring isn't clear enough for me to understand what it was even supposed to do in the first place. I would just delete the function